### PR TITLE
Not able to detect SE_NO_SERIALVERSIONID bug - added unit test

### PR DESCRIPTION
**Pull request by yuliyasokh**
----

Make sure these boxes are checked before submitting your PR -- thank you!

- [v] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

File: spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue2436Test.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import static edu.umd.cs.findbugs.test.CountMatcher.containsExactly;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class Issue2436Test extends AbstractIntegrationTest {
+
+    @Test
+    public void test() {
+        performAnalysis("lambdas/Issue2436.class");
+        BugInstanceMatcher bugMatcher = new BugInstanceMatcherBuilder().build();
```
Comment (JuditKnoll) [inline spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue2436Test.java:16]: Can you please add at least a bugtype to the BugInstanceMatcher?

File: spotbugsTestCases/src/java/lambdas/Issue2436.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package lambdas;
```
Comment (JuditKnoll) [inline spotbugsTestCases/src/java/lambdas/Issue2436.java:1]: This issue is not about lambas, can you please move this file to the appropriate package? If you can't find better, then `ghIssues` would be okay. (Don't forget to modify the path in the test file as well.)

**Review by JuditKnoll — CHANGES_REQUESTED** [review]

File: spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue2436Test.java (hunk 1 above)
Comment (yuliyasokh) [reply spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue2436Test.java]: thanks, added bugtype and confidence

File: spotbugsTestCases/src/java/ghIssues/Issue2436.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package ghIssues;
+
+import java.io.Serializable;
+
+public class Issue2436 implements Serializable {
+    public void doNothing(){
```
Comment (JuditKnoll) [inline spotbugsTestCases/src/java/ghIssues/Issue2436.java:6]: Is this function really needed for a minimal example?

File: spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue2436Test.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+public class Issue2436Test extends AbstractIntegrationTest {
+
+    @Test
+    public void test() {
+        performAnalysis("ghIssues/Issue2436.class");
+        BugInstanceMatcher bugMatcher = new BugInstanceMatcherBuilder()
+                .bugType("SE_NO_SERIALVERSIONID")
+                .withConfidence(Confidence.LOW)
+                .build();
```
Comment (JuditKnoll) [inline spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue2436Test.java:20]: There are usually two ways when we use BugInstanceMatcher:
1. negative tests: the BugInstanceMatcher should contain minimal info (only the bugtype, maybe the containing class), used when we want to assert the number of hits of a certain bugtype, or we want to check that a bug is not reported at all.
2. positive tests: the BugInstanceMatcher should contain maximal info, used when we want to assert that a bug is reported in a certain way exactly.

IMO adding the confidence makes it closer to the positive tests, than to the negative, general ones.

File: spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue2436Test.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        assertThat(getBugCollection(), containsExactly(1, bugMatcher));
+    }
+
+    }
```
Comment (JuditKnoll) [inline spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue2436Test.java:24]: Can you please modify the indenting to be conform to the project? Or you can simply run `./gradlew spotlessApply` or more specifically the `./gradlew :spotbugs-tests:spotlessApply` command as you can see in the build fail.

**Review by JuditKnoll — CHANGES_REQUESTED** [review]
Please make sure that the build is successful when creating your PR.

File: spotbugsTestCases/src/java/ghIssues/Issue2436.java (hunk 3 above)
Comment (yuliyasokh) [reply spotbugsTestCases/src/java/ghIssues/Issue2436.java]: true, removed it

File: spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue2436Test.java (hunk 4 above)
Comment (yuliyasokh) [reply spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue2436Test.java:18]: ok, got it

**Review by JuditKnoll — APPROVED** [review]

**Comment by yuliyasokh** [comment]
@JuditKnoll could you please merge this request as I do not have write access?
